plugin/types.go

```diff
@@ -235,3 +237,25 @@ type Signalable interface {
 	Entry
 	Signal(context.Context, string) error
 }
+
+// This interface exists to break the circular dependency between plugin and external.
+// The external plugin implementation is in its own module so it can use other modules
+// that implement new features and have dependencies on this module.
+type externalPlugin interface {
+	MethodSupported(string) MethodSignature
```

Comment:
Why not `MethodSignature` ? `MethodSupported` seems like a weird name.

Reply:
Inherited from SupportedMethods, but MethodSignature seems reasonable too.

Reply:
Let's go with that then.